apps/gitness/src/pages/project-settings/project-settings-sandbox-page.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState, useEffect } from 'react'
+import { Button, ButtonGroup, Input, Spacer, Text, Icon } from '@harnessio/canary'
+import { SandboxLayout, FormFieldSet, MessageTheme } from '@harnessio/playground'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { z } from 'zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { FormDialogProjectDelete } from './components/FormDialogProjectDelete'
+
+interface PageProps {
+  spaceData: InputProps
+  onFormSubmit: (formData: InputProps) => void
+  onHandleDescription: (newDescription: string) => void
+  handleDeleteProject: () => void
+  isDeleteSuccess: boolean
+  isDeleting: boolean
+  updateError: string | null
+  deleteError: string | null
+}
+interface InputProps {
+  identifier: string
+  description: string
+}
+
+// Define form schema for Project Settings
+const projectSettingsSchema = z.object({
+  identifier: z.string().min(1, { message: 'Please provide a project name' }),
+  description: z.string().min(1, { message: 'Please provide an description' })
+})
+
+// Define TypeScript type
+type ProjectSettingsFields = z.infer<typeof projectSettingsSchema>
+
+export const ProjectSettingsSandboxPage = ({
+  onFormSubmit,
+  spaceData,
+  handleDeleteProject,
+  isDeleteSuccess,
+  isDeleting,
+  updateError,
+  deleteError
+}: PageProps) => {
+  // Project Settings form handling
+  const {
+    register,
+    handleSubmit,
+    // TODO: will use this to reset the form after api call has projectName
+    // reset: resetProjectSettingsForm,
+    resetField,
+    setValue,
+    formState: { errors, isValid, isDirty }
+  } = useForm<ProjectSettingsFields>({
+    resolver: zodResolver(projectSettingsSchema),
+    mode: 'onChange',
+    defaultValues: {
+      identifier: spaceData?.identifier, //project name
+      description: spaceData?.description
+    }
+  })
+
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [submitted, setSubmitted] = useState(false)
```

Comment:
I think instead of trying to manage `submitting` state on your own you could utilize `react-queries` out-of-box functionalities

Reply:
Noted & will refactored in follow up PR